### PR TITLE
feat: Add user input page with form validation and Tailwind CSS styling

### LINKED ISSUE #2: User Input Page
### Issue Description

Create a form that allows users to specify the number of floors (1-100) and lifts (1-10), and trigger the simulation.
### Expected Behavior

- Users should be able to specify the number of floors (1-100) and lifts (1-10).
- Users should be able to trigger the simulation after providing inputs.
- Inputs must be validated to prevent negative or out-of-range values, with clear error messages for invalid inputs.

### Severity/Priority
 - [x] Critical
 - [ ] High
 - [ ] Medium
 - [ ] Low

### Checklist

- [ ] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [ ] I have provided all the necessary information to understand and reproduce the issue.
- [ ] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by Achintya-Chatterjee**
Date: `25th September, 2024`

Developer Name: @Achintya-Chatterjee 

---

## Issue Ticket Number

 - Closes #2 
## Description
I created the user input page for the lift simulation. This page allows users to specify the number of floors (1-100) and lifts (1-10) and form validation to prevent invalid inputs. Tailwind CSS is used to ensure responsive design and styling.

- Implemented a form for users to input the number of floors and lifts.
- Added validation for floor (1-100) and lift (1-10) inputs.
- Styled the input form using Tailwind CSS to ensure responsiveness across devices.
- Displayed error messages for invalid inputs (e.g., negative numbers, out-of-range values).

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

https://github.com/user-attachments/assets/7d0e7135-47f6-4e96-8214-05d65139fc4e

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-09-25 at 21 03 12

</details>

### CODE REVIEW

**Review by rishirishhh — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,70 @@
...
+  const [lifts, setLifts] = useState<number | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const handleStartSimulation = () => {
+    if (!floors || !lifts || floors < 1 || floors > 100 || lifts < 1 || lifts > 10) {
+      setError('Please enter valid values (Floors: 1-100, Lifts: 1-10)');
+    } else {
+      setError(null);
+      // Proceed with simulation logic
+      console.log('Starting simulation...');
```
Comment (tejaskh3) [inline src/App.tsx:14]: can we remove this console?

File: src/__tests__/App.test.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+describe('App component', () => {
+  it('should render the form inputs and button', () => {
+    render(<App />);
+
+    expect(screen.getByLabelText(/Number of Floors/i)).toBeInTheDocument();
+    expect(screen.getByLabelText(/Number of Lifts/i)).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: /Start Simulation/i })).toBeInTheDocument();
+  });
+
+  it('should show an error message for invalid input values', () => {
```
Comment (tejaskh3) [inline src/__tests__/App.test.tsx:17]: can you please add test cases for below the given range as well?

File: src/App.tsx (hunk 1 above)
Comment (Achintya-Chatterjee) [reply src/App.tsx:14]: will remove the next release

File: src/__tests__/App.test.tsx (hunk 2 above)
Comment (Achintya-Chatterjee) [reply src/__tests__/App.test.tsx:17]: added, thanks for pointing this out

**Pull request merged**